### PR TITLE
Chore/show rr UI only if project is enabled

### PR DESCRIPTION
**Pull request by joshenlim**
- Add DatabaseSelector to API settings and show corresponding replica API endpoint in settings/api if selected
- Only show RR UI if enabled on a per project level
- Shift DatabaseSelector state to valtio
- Deprecate `ProjectBase` and `Project` interface from types, use type from codegen as source of truth

### CODE REVIEW

**Comment by joshenlim** [comment]
Note to self once im ready to jump back on this - Change project store to database selector store (or something that sounds nicer)

File: apps/studio/components/interfaces/Settings/Infrastructure/InfrastructureActivity.tsx [hunk 1]
```diff
@@ -188,12 +188,7 @@ const InfrastructureActivity = () => {
       </ScaffoldContainer>
       <ScaffoldContainer className="sticky top-0 py-6 border-b bg-background z-10">
         <div className="flex items-center gap-x-4">
-          {/* {readReplicasEnabled && (
-            <DatabaseSelector
-              selectedDatabaseId={selectedDatabaseId}
-              onChangeDatabaseId={setSelectedDatabaseId}
-            />
-          )} */}
+          {/* {readReplicasEnabled && <DatabaseSelector />} */}
```
Comment (alaister) [inline apps/studio/components/interfaces/Settings/Infrastructure/InfrastructureActivity.tsx:191]: Still want this commented out?

File: apps/studio/components/layouts/ProjectLayout/LayoutHeader/NotificationsPopover/NotificationRow.tsx [hunk 2]
```diff
@@ -3,18 +3,18 @@ import dayjs from 'dayjs'
...
-import { useOrganizationsQuery } from 'data/organizations/organizations-query'
 
 export interface NotificationRowProps {
   notification: Notification
-  onSelectRestartProject: (project: Project, notification: Notification) => void
-  onSelectApplyMigration: (project: Project, notification: Notification) => void
-  onSelectRollbackMigration: (project: Project, notification: Notification) => void
+  onSelectRestartProject: (project: any, notification: Notification) => void
+  onSelectApplyMigration: (project: any, notification: Notification) => void
+  onSelectRollbackMigration: (project: any, notification: Notification) => void
```
Comment (alaister) [inline apps/studio/components/layouts/ProjectLayout/LayoutHeader/NotificationsPopover/NotificationRow.tsx:17]: Feels like the TS went backwards here – can we use ProjectInfo now or something similar?

File: apps/studio/components/layouts/ProjectLayout/ProjectLayout.tsx [hunk 3]
```diff
@@ -201,6 +205,10 @@ const ContentWrapper = ({ isLoading, isBlocking = true, children }: ContentWrapp
     selectedProject?.status === PROJECT_STATUS.PAUSING
   const isProjectOffline = selectedProject?.postgrestStatus === 'OFFLINE'
 
+  useEffect(() => {
+    if (ref) state.setSelectedDatabaseId(ref)
+  }, [ref])
```
Comment (alaister) [inline apps/studio/components/layouts/ProjectLayout/ProjectLayout.tsx:210]: Is this still needed now that it's in a context?

File: apps/studio/data/projects/project-detail-query.ts [hunk 4]
```diff
@@ -1,18 +1,34 @@
...
+export type ProjectMinimal = components['schemas']['ProjectInfo']
+export type ProjectDetail = components['schemas']['ProjectDetailResponse']
+
+export interface Project extends ProjectDetail {
+  /**
+   * postgrestStatus is available on client side only.
+   * We use this status to check if a project instance is HEALTHY or not
+   * If not we will show ConnectingState and run a polling until it's back online
+   */
+  postgrestStatus?: 'ONLINE' | 'OFFLINE'
```
Comment (alaister) [inline apps/studio/data/projects/project-detail-query.ts:19]: > available on client side only

Where is it being added in?

**Review by alaister — COMMENTED** [review]
Looking great!

File: apps/studio/components/interfaces/Settings/Infrastructure/InfrastructureActivity.tsx (hunk 1 above)
Comment (joshenlim) [reply apps/studio/components/interfaces/Settings/Infrastructure/InfrastructureActivity.tsx:191]: Yup! activity metrics for replicas are still not done, i'll leave a comment actually

File: apps/studio/components/layouts/ProjectLayout/ProjectLayout.tsx (hunk 3 above)
Comment (joshenlim) [reply apps/studio/components/layouts/ProjectLayout/ProjectLayout.tsx:210]: yup this is to initialize the value in the valtio store - unless there's a better way we can do this?

File: apps/studio/data/projects/project-detail-query.ts (hunk 4 above)
Comment (joshenlim) [reply apps/studio/data/projects/project-detail-query.ts:19]: when we call `setProjectPostgrestStatus` from projects-query

**Review by alaister — APPROVED** [review]
LGTM!

**Pull request merged**